llvm/lib/Target/NVPTX/NVPTXAliasAnalysis.h

```diff
@@ -36,6 +36,12 @@ class NVPTXAAResult : public AAResultBase {
 
   ModRefInfo getModRefInfoMask(const MemoryLocation &Loc, AAQueryInfo &AAQI,
                                bool IgnoreLocals);
+
+  MemoryEffects getMemoryEffects(const CallBase *Call, AAQueryInfo &AAQI);
+
+  MemoryEffects getMemoryEffects(const Function *F) {
+    return MemoryEffects::unknown();
+  }
```

Comment:
I'm not super familiar with AA. What's the default here? Is it not [this](https://github.com/llvm/llvm-project/blob/8c7f0eaa6ee3f84e3d8260535cced234bed4fa28/llvm/include/llvm/Analysis/AliasAnalysis.h#L850-L852)?

Reply:
This is needed for C++ subclassing reasons, I don't remember (understand) the specifics. Without it there is a build error.